disnake/app_commands.py

```diff
@@ -336,59 +336,53 @@ class ApplicationCommand(ABC):
         Whether this command can be used in DMs.
     """
 
-    __repr_info__: ClassVar[Tuple[str, ...]] = ("type", "name")
+    __repr_info__: ClassVar[Tuple[str, ...]] = (
+        "type",
+        "name",
+        "dm_permission",
+        "default_member_permisions",
+    )
 
     def __init__(
         self,
         type: ApplicationCommandType,
         name: str,
-        default_permission: bool = MISSING,
-        dm_permission: bool = MISSING,
-        default_member_permissions: Optional[Permissions] = MISSING,
+        dm_permission: bool = True,
+        default_member_permissions: Optional[Union[Permissions, int]] = None,
     ):
         self.type: ApplicationCommandType = enum_if_int(ApplicationCommandType, type)
         self.name: str = name
 
-        if default_permission is not MISSING:
-            warn_deprecated(
-                "default_permission is deprecated. "
-                "Please use dm_permission and default_member_permissions instead",
-                DeprecationWarning,
-                stacklevel=2,
-            )
-            if default_permission:
-                self.dm_permission = True
-                default_member_permissions = None
-            else:
-                self.dm_permission = False
-                default_member_permissions = Permissions(0)
-
-        if dm_permission is not MISSING:
-            self.dm_permission: bool = dm_permission
+        self.dm_permission: bool = dm_permission
 
         self._default_member_permissions: Optional[int]
         if default_member_permissions is None:
             # allow everyone to use the command if its not supplied
             self._default_member_permissions = None
+        elif isinstance(default_member_permissions, int):
+            self._default_member_permissions = default_member_permissions
         else:
             self._default_member_permissions = default_member_permissions.value
 
         self._always_synced: bool = False
 
     @property
     def default_member_permissions(self) -> Optional[Permissions]:
-        """Optional[:class:`Permissions`]: The default member permissions for this command.
+        """Optional[:class:`Permissions`]: The default required member permissions for this command.
+        A member must have *all* these permissions to be able to invoke the command in a guild.
```

Comment:
I thought it was any.

Reply:
due to a dysfunctional command cache in my test server i am unable to test this right now.

Reply:
I tested it, and it does require you to have *all* permissions, not just any one of them.

Also see https://msciotti.notion.site/msciotti/Command-Permissions-V2-4d113cb49090409f998f3bd80a06c3bd:
>Example: You create a /secret command with default_permission: 536872961. When created, Discord will disable /ban [sic] for everyone except the roles that have the "Create Instant Invite", "Manage Webhooks", and "Send Messages" permission.